File: sparc-backend/utils/graphPresets.js
// Graph presets based on standard algorithmic testing scenarios

const presets = {
  small: {
    nodes: Array.from({ length: 5 }, (_, i) => ({ id: i, label: String(i) })),
    edges: [
      { source: 0, target: 1, weight: 10 },
      { source: 0, target: 3, weight: 30 },
      { source: 0, target: 4, weight: 100 },
      { source: 1, target: 2, weight: 50 },
      { source: 2, target: 4, weight: 10 },
      { source: 3, target: 2, weight: 20 },
      { source: 3, target: 4, weight: 60 }
    ],
    directed: true
  },
  medium: {
    nodes: Array.from({ length: 9 }, (_, i) => ({ id: i, label: String(i) })),
    edges: [
      { source: 0, target: 1, weight: 4 },
      { source: 0, target: 7, weight: 8 },
      { source: 1, target: 2, weight: 8 },
      { source: 1, target: 7, weight: 11 },
      { source: 2, target: 3, weight: 7 },
      { source: 2, target: 8, weight: 2 },
      { source: 2, target: 5, weight: 4 },
      { source: 3, target: 4, weight: 9 },
      { source: 3, target: 5, weight: 14 },
      { source: 4, target: 5, weight: 10 },
      { source: 5, target: 6, weight: 2 },
      { source: 6, target: 7, weight: 1 },
      { source: 6, target: 8, weight: 6 },
      { source: 7, target: 8, weight: 7 }
    ],
    directed: true
  },
  negative: {
    nodes: Array.from({ length: 4 }, (_, i) => ({ id: i, label: String(i) })),
    edges: [
      { source: 0, target: 1, weight: 4 },
      { source: 0, target: 3, weight: 5 },
      { source: 1, target: 2, weight: -5 }, // Negative weight
      { source: 2, target: 3, weight: 3 }
    ],
    directed: true
  },
  dense: {
    nodes: Array.from({ length: 6 }, (_, i) => ({ id: i, label: String(i) })),
    edges: [
      { source: 0, target: 1, weight: 2 }, { source: 0, target: 2, weight: 4 }, { source: 0, target: 3, weight: 1 }, { source: 0, target: 4, weight: 8 }, { source: 0, target: 5, weight: 3 },
      { source: 1, target: 2, weight: 1 }, { source: 1, target: 3, weight: 7 }, { source: 1, target: 4, weight: 2 }, { source: 1, target: 5, weight: 5 },
      { source: 2, target: 3, weight: 3 }, { source: 2, target: 4, weight: 6 }, { source: 2, target: 5, weight: 1 },
      { source: 3, target: 4, weight: 5 }, { source: 3, target: 5, weight: 4 },
      { source: 4, target: 5, weight: 2 }
    ],
    directed: true
  }
};

module.exports = presets;

File: backend/GraphPresets.java
import java.util.HashMap;
import java.util.Map;

public class GraphPresets {
    
    public static Map<String, Graph> presets = new HashMap<>();

    static {
        Graph small = new Graph();
        for (int i = 0; i < 5; i++) small.nodes.add(new Graph.Node(i));
        small.edges.add(new Graph.Edge(0, 1, 10));
        small.edges.add(new Graph.Edge(0, 4, 100));
        small.edges.add(new Graph.Edge(1, 2, 50));
        small.edges.add(new Graph.Edge(2, 4, 10));
        small.edges.add(new Graph.Edge(1, 4, 50));
        small.edges.add(new Graph.Edge(0, 3, 30));
        small.edges.add(new Graph.Edge(3, 4, 60));
        presets.put("small", small);

        Graph medium = new Graph();
        for (int i = 0; i < 9; i++) medium.nodes.add(new Graph.Node(i));
        medium.edges.add(new Graph.Edge(0, 1, 4));
        medium.edges.add(new Graph.Edge(0, 7, 8));
        medium.edges.add(new Graph.Edge(1, 2, 8));
        medium.edges.add(new Graph.Edge(1, 7, 11));
        medium.edges.add(new Graph.Edge(2, 3, 7));
        medium.edges.add(new Graph.Edge(2, 8, 2));
        medium.edges.add(new Graph.Edge(2, 5, 4));
        medium.edges.add(new Graph.Edge(3, 4, 9));
        medium.edges.add(new Graph.Edge(3, 5, 14));
        medium.edges.add(new Graph.Edge(4, 5, 10));
        medium.edges.add(new Graph.Edge(5, 6, 2));
        medium.edges.add(new Graph.Edge(6, 7, 1));
        medium.edges.add(new Graph.Edge(6, 8, 6));
        medium.edges.add(new Graph.Edge(7, 8, 7));
        presets.put("medium", medium);

        Graph dense = new Graph();
        for (int i = 0; i < 6; i++) dense.nodes.add(new Graph.Node(i));
        for (int i = 0; i < 6; i++) {
            for (int j = i + 1; j < 6; j++) {
                int w = (int) (Math.random() * 50) + 1;
                dense.edges.add(new Graph.Edge(i, j, w));
            }
        }
        presets.put("dense", dense);

        Graph negative = new Graph();
        for (int i = 0; i < 5; i++) negative.nodes.add(new Graph.Node(i));
        negative.edges.add(new Graph.Edge(0, 1, -1));
        negative.edges.add(new Graph.Edge(0, 2, 4));
        negative.edges.add(new Graph.Edge(1, 2, 3));
        negative.edges.add(new Graph.Edge(1, 3, 2));
        negative.edges.add(new Graph.Edge(1, 4, 2));
        negative.edges.add(new Graph.Edge(3, 2, 5));
        negative.edges.add(new Graph.Edge(3, 1, 1));
        negative.edges.add(new Graph.Edge(4, 3, -3));
        presets.put("negative", negative);
    }
}
